modules/photons_control/device_finder.py

```diff
@@ -541,13 +576,86 @@ def location_id(self):
             return sb.NotSpecified
         return self.location.uuid
 
+    @property
+    def product_id(self):
+        if self.product is sb.NotSpecified:
+            return sb.NotSpecified
+        return self.product.pid
+
+    @property
+    def product_name(self):
+        if self.product_id is sb.NotSpecified:
+            return sb.NotSpecified
+        return self.product.friendly
+
+    @property
+    def product_type(self):
+        if self.product is not sb.NotSpecified:
+            return (
+                ProductType.LIGHT.value
+                if Products[VendorRegistry.LIFX, self.product.pid].cap.is_light
+                else ProductType.SWITCH.value
+            )
+        else:
+            return sb.NotSpecified
+
+    @property
+    def firmware_version(self):
+        if self.firmware is sb.NotSpecified:
+            return sb.NotSpecified
+        return str(self.firmware)
```

Comment:
I'd say remove your `__str__` method and return `tuple(self.firmware)` and add an

```
def __iter__(self):
   yield from (self.major, self.minor)
```
<img width="495" alt="Screen Shot 2022-01-02 at 9 18 20 pm" src="https://user-images.githubusercontent.com/109301/147872794-cb529d2e-4e74-4a1c-85c3-884febe9d9b0.png">


To the firmware object

because firmware version is a tuple of two numbers and should be treated as such.